package.json

```diff
@@ -40,49 +40,55 @@
     "@nuxtjs/i18n": "^7.0.3",
     "@nuxtjs/sentry": "^5.0.3",
     "@nuxtjs/sitemap": "^2.4.0",
+    "@nuxtjs/style-resources": "^1.0.0",
+    "@nuxtjs/eslint-module": "^3.0.2",
     "@nuxtjs/svg": "^0.1.12",
-    "axios": "^0.21.1",
+    "@nuxtjs/tailwindcss": "^4.2.1",
```

Comment:
Tailwind also should be a dev-dependency, the build step will generate the actual CSS used in prod.

Reply:
Same for PostCSS and Sass.

Reply:
There's an unclear distinction with 'devDependencies' on whether they should be 'deps not included in the production bundle' _or_ 'deps needed for local development'. to put the question differently:

If we run `npm install --production && npm run build`, should the build work?

cc: @WordPress/openverse-frontend

Reply:
Well, when you add in an SSR server that needs to be able to build things inside the container, the distinction gets muddied. Production dependencies should include everything required to "run the app." If we built a standalone package with styles that docker was able to consume then yeah, tailwind would be a dev dependency because the only production dependency would be its output. But because our "production" release includes building the app from scratch inside the container then we need to include some of these dependencies that seem to cross the boundaries.

Reply:
I agree 👍

Reply:
> If we run npm install --production && npm run build, should the build work?

From a DevOps perspective, it shouldn't.
Build your application is one step before getting it up and running. Being very simplistic, production doesn't care how you came to create a distribution package.
They are only interested in having the necessary files and commands to be able to run freely and independently.

So the question we should make is: **what packages should be placed in dependencies to make sure to run the app as soon as I have the dist folder ready**, that's how I consider `nuxt` should be one of the dependencies (because we need to run `nuxt start`) but `husky` shouldn't.

In addition, that's why I'm using two stages in `Dockerfile`. The first stage requires `devDependencies` to build the application, but, the Production stage doesn't them, that's why I'm transferring the `dist` folder from one stage to the other, because that's the only assets I'm care of.

More info: https://www.docker.com/blog/speed-up-your-development-flow-with-these-dockerfile-best-practices/

Reply:
Definitely agree that `husky` shouldn't be a regular dependency. Nothing it does is necessary for building the app, it's just meant to run developer level verification tools.

If you're doing a two-stage docker build like that then yeah, you can and should move anything that isn't necessary for _after_ `nuxt build` is run I guess.

Reply:
I like how the approach of @rbadillap sounds with two stages build, keeping the running production environment lean. Moving dependencies around I got this warning message in one opportunity:

>  WARN  Module @nuxtjs/tailwindcss not found. Please ensure @nuxtjs/tailwindcss is in devDependencies and installed. HINT: During build step, for npm/yarn, NODE_ENV=production or --production should NOT be used.